server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -279,9 +302,28 @@ public ActivationKey removeProductIdFromKey(
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<ActivationKey> findActivationKey() {
-        List<ActivationKey> keyList = activationKeyCurator.listAll();
-        return keyList;
+    public Response findActivationKey() {
+        // TODO: Replace this with use of a cursor/iterator
```

Comment:
@Ceiu The cursor stuff is in #1215 right? Do you want to wait for it to hit the master?